Libraries/JavaScriptAppEngine/Initialization/exceptionOccurHint.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule exceptionOccurHint
+ */
+
+'use strict';
+
+const SourceMapsCache = require('SourceMapsCache');
+const parseErrorStack = require('parseErrorStack');
+
+function exceptionOccurHint() {
+  const stack =  parseErrorStack(new Error());
+  const renderFrame = stack.find((frame) => frame.methodName.endsWith('render'));
```

Comment:
Oh sorry, I see - you don't want to report the line where the warning was fired since that's in the infra code...maybe we could just give the line of the warning by default, but have some custom logic for the props case you're focused on here - does that make sense? Basically if the stack looks like it comes from the strict prop type checker then we grab the render function, otherwise we just show the file/line of the actual warning.

Also, we could just include the entire parsed error stack at the bottom of the warning message.